main.go

```diff
@@ -0,0 +1,153 @@
+	"encoding/base64"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/rarimo/certificate-transparency-go/asn1"
+	"github.com/rarimo/certificate-transparency-go/x509"
+
+	"github.com/github/smimesign/ietf-cms/protocol"
+)
+
+func ReadLDIF(fileName string) ([]LDIFRawData, error) {
+	fileInfo, err := os.Stat(fileName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file stat: %w", err)
+	}
+
+	ldifFile, err := os.OpenFile(fileName, os.O_RDONLY, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %w", err)
+	}
+	defer ldifFile.Close()
+
+	ldifData := make([]byte, fileInfo.Size())
+	_, err = ldifFile.Read(ldifData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %w", err)
+	}
+
+	re, err := regexp.Compile(`(?s)cn: (.+?)\n(?:objectClass:.*?\n)+?.*?pkdMasterListContent:: (.*?)\n\n`)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile regexp: %w", err)
+	}
+
+	dirtyLDIFData := re.FindAllStringSubmatch(string(ldifData), -1)
+	ldifRawData := make([]LDIFRawData, len(dirtyLDIFData))
+	for i, entry := range dirtyLDIFData {
+		ldifRawData[i].CN = strings.ReplaceAll(entry[1], "\n ", "")
+
+		dataB64 := strings.ReplaceAll(entry[2], "\n ", "")
+		data, err := base64.StdEncoding.DecodeString(dataB64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode MasterListContent: %w", err)
+		}
+		ldifRawData[i].Data = data
+	}
+
+	return ldifRawData, nil
+}
+
+func ReadLDIF2(fileName string) ([][]byte, error) {
+	fileInfo, err := os.Stat(fileName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file stat: %w", err)
+	}
+
+	ldifFile, err := os.OpenFile(fileName, os.O_RDONLY, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %w", err)
+	}
+	defer ldifFile.Close()
+
+	ldifData := make([]byte, fileInfo.Size())
+	_, err = ldifFile.Read(ldifData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %w", err)
+	}
+
+	re, err := regexp.Compile(`(?s)pkdMasterListContent:: (.*?)\n\n`)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile regexp: %w", err)
+	}
+
+	dirtyLDIFData := re.FindAllStringSubmatch(string(ldifData), -1)
+	ldifRawData := make([][]byte, len(dirtyLDIFData))
+	for i, entry := range dirtyLDIFData {
+		dataB64 := strings.ReplaceAll(entry[1], "\n ", "")
+		data, err := base64.StdEncoding.DecodeString(dataB64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode MasterListContent: %w", err)
+		}
+		ldifRawData[i] = data
+	}
+
+	return ldifRawData, nil
+}
+
+func ExtractMasterLists(rawData [][]byte) (mls []CSCAMasterList, err error) {
+	mls = make([]CSCAMasterList, len(rawData))
+	for i, entry := range rawData {
+		derData, err := protocol.BER2DER(entry)
+		if err != nil {
+			return nil, err
+		}
+
+		var content ContentInfo
+		_, err = asn1.Unmarshal(derData, &content)
+		if err != nil {
+			return nil, err
+		}
+
+		var cscaMasterList CSCAMasterList
+		_, err = asn1.Unmarshal(content.Content.EncapContInfo.EContent, &cscaMasterList)
```

Comment:
There is `SignedDataContent()` method in response from previous comment, use it to unmarshal into CSCAMasterList